dace/codegen/CMakeLists.txt

```diff
@@ -512,6 +566,64 @@ if(DACE_ENABLE_XILINX)
 
 endif() # DACE_ENABLE_XILINX
 
+# create verilator RTL simulation objects
+if(DACE_ENABLE_RTL)
+
+  # find verilator installation
+  find_package(verilator HINTS $ENV{VERILATOR_ROOT} ${VERILATOR_ROOT})
+  if (NOT verilator_FOUND)
+    message(FATAL_ERROR "Verilator was not found. Either install it, or set the VERILATOR_ROOT environment variable")
+  endif()
+
+  # check minimal version requirements
+  set(VERILATOR_MIN_VERSION "4.028")
+  if("${verilator_VERSION}" VERSION_LESS VERILATOR_MIN_VERSION)
+    message(WARNING "Please upgrade verilator to version >=${VERILATOR_MIN_VERSION}")
+  endif()
+
+  # get verilator flags from dace.conf
+  set(VERILATOR_FLAGS "${DACE_RTL_VERILATOR_FLAGS}")
+
+  # add lint verilator flags
+  if("${DACE_RTL_VERILATOR_LINT_WARNINGS}")
+    # -Wall: Enable all style warnings
+    # -Wno-fatal: Disable fatal exit on warnings
+    set(VERILATOR_FLAGS "${VERILATOR_FLAGS}" "-Wall" "-Wno-fatal")
+  endif()
+
+  message(WARNING "${DACE_RTL_FILES}")
+  foreach(RTL_FILE ${DACE_RTL_FILES})
+
+    # extract design name
+    get_filename_component(RTL_FILE_NAME "${RTL_FILE}" NAME_WE)
+
+    # add verilated .cpp files
+    set(VERILATOR_SRC "${dace_program_BINARY_DIR}/CMakeFiles/${RTL_FILE_NAME}.dir/V${RTL_FILE_NAME}.dir/V${RTL_FILE_NAME}.cpp")
+    set(VERILATOR_SRC "${VERILATOR_SRC}" "${dace_program_BINARY_DIR}/CMakeFiles/${RTL_FILE_NAME}.dir/V${RTL_FILE_NAME}.dir/V${RTL_FILE_NAME}__Syms.cpp")
+    set(VERILATOR_SRC "${VERILATOR_SRC}" "${VERILATOR_ROOT}/include/verilated.cpp")
+    set(DACE_CPP_FILES "${DACE_CPP_FILES}" "${VERILATOR_SRC}")
```

Comment:
Does this not cause the C++ files to be re-compiled as part of the main DaCe shared library object?

Reply:
Generally: I wouldn't call it re-compilation, but yes, they will be compiled into the DaCe library (this is intended)
For line 601 and 602: The verilator cmake command `verilate` basically creates these .cpp files (and does not compile them), which are the cpp equivalent of the verilator design. We need to have them in the binary file, such that we can call them from the tasklet cpp code location.
For line 603: I agree with you that this lines might be added multiple times (for multi rtl tasklets). I am going to move it ouside of the loop.